subiquity/server/controllers/filesystem.py

```diff
@@ -375,7 +376,12 @@ async def v2_add_partition_POST(self, data: AddPartitionV2) \
             'mount': data.partition.mount,
         }
 
-        self.create_partition(disk, data.gap, spec, wipe='superblock')
+        flag = ""
+        if data.partition.partition_type in ('extended', 'logical'):
+            flag = data.partition.partition_type
+
```

Comment:
Something somewhere should yell if you try to add an extended partition to a gpt disk I think.

Also, on the server side a Gap knows if it's in an extended partition, so in a sense sending partition_type == logical is redundant. Maybe we should extend the Gap that goes across the API to have that and then just have a boolean field to indicate creating an extended partition?